### PR TITLE
feat: care taker availability

### PR DESCRIPTION
**Pull request by ambhinav**
Created a skeleton for **care taker availability**

**Schema**
- `care_taker_part_timers_available_dates`
- `care_taker_full_timers_unavailable_dates`
- Check constraints for both
- `DATE` type used to store availability

**Service**
- Added to `UserInfoService`

**Endpoints**: `/user/care-taker/...`
- Added to `UserController`

### CODE REVIEW

File: src/User/UserController.js [hunk 1]
```diff
@@ -265,6 +265,74 @@ app.post(
...
+  [
+    body('type').isIn([
+      RoleUtils.CARE_TAKER_FULL_TIMER,
+      RoleUtils.CARE_TAKER_PART_TIMER,
+    ]),
+  ],
+  AuthRequired,
+  async (req, res) => {
+    try {
+      const response = await service.UserCareTakerAvailabilityDateCreate({
```
Comment (ArunBeCoding) [inline src/User/UserController.js]: Why is the service call UserCareTakerAvailabilityDateCreate? Should be info right

File: setup/CreateDB.sql [hunk 2]
```diff
@@ -90,6 +91,245 @@ CREATE TABLE care_taker_part_timers (
...
+  ELSE
+    RETURN true;
+  END IF;
+END;
+$$
+LANGUAGE 'plpgsql';
+
+CREATE TABLE care_taker_full_timers_unavailable_dates (
+  email VARCHAR REFERENCES care_taker_full_timers(email),
+  date DATE NOT NULL,
```
Comment (nwjbrandon) [inline setup/CreateDB.sql]: Can explain why the date can be nullable?

File: setup/CreateDB.sql [hunk 3]
```diff
@@ -90,6 +91,245 @@ CREATE TABLE care_taker_part_timers (
...
+  CHECK(check_care_taker_availability(email, date)),
+  PRIMARY KEY(email, date)
+);
+
+CREATE TABLE care_taker_part_timers_available_dates (
+  email VARCHAR REFERENCES care_taker_part_timers(email),
+  date DATE NOT NULL,
```
Comment (nwjbrandon) [inline setup/CreateDB.sql]: Can explain why the date can be nullable?

File: setup/CreateDB.sql (hunk 2 above)
Comment (ambhinav) [reply setup/CreateDB.sql]: yep you're right its redundant since its part of the PK. I will update, thanks.

**Pull request merged**